opentracing-spring-web-handler-interceptor/src/test/java/io/opentracing/contrib/spring/web/servlet/TracingHandlerInterceptorTest.java

```diff
@@ -0,0 +1,256 @@
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.jayway.awaitility.Awaitility;
+
+import io.opentracing.mock.MockSpan;
+import io.opentracing.mock.TextMapPropagator;
+import io.opentracing.tag.Tags;
+
+/**
+ * @author Pavol Loffay
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = {AppConfiguration.class,
+    WebSecurityConfig.class})
+public class TracingHandlerInterceptorTest {
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @LocalServerPort
+    private int port;
+
+    @Before
+    public void beforeTest() throws InterruptedException {
+        AppConfiguration.mockTracer.reset();
+    }
+
+    @Test
+    public void contextPropagation() throws Exception {
+        {
+            HttpHeaders headers = new HttpHeaders();
+            headers.set(TextMapPropagator.HEADER_SPAN_ID, "1");
+            headers.set(TextMapPropagator.HEADER_TRACE_ID, "345");
+
+            HttpEntity<String> entity = new HttpEntity<>(headers);
+
+            restTemplate.exchange("/sync", HttpMethod.GET, entity, String.class);
+            Awaitility.await().until(reportedSpansSize(), IsEqual.equalTo(1));
+        }
+
+        List<MockSpan> mockSpans = AppConfiguration.mockTracer.finishedSpans();
+        Assert.assertEquals(1, mockSpans.size());
+
+        MockSpan span = mockSpans.get(0);
+        Assert.assertEquals(1, span.parentId());
+        Assert.assertEquals(345, span.context().traceId());
+    }
+
+    @Test
+    public void testSyncWithStandardTags() throws Exception {
+        ResponseEntity<String> response = restTemplate.getForEntity("/sync", String.class);
+        Awaitility.await().until(reportedSpansSize(), IsEqual.equalTo(1));
+
+        List<MockSpan> mockSpans = AppConfiguration.mockTracer.finishedSpans();
+        Assert.assertEquals(1, mockSpans.size());
+
+        MockSpan span = mockSpans.get(0);
+        Assert.assertEquals("GET", span.operationName());
+        Assert.assertEquals(5, span.tags().size());
+        Assert.assertEquals(Tags.SPAN_KIND_SERVER, span.tags().get(Tags.SPAN_KIND.getKey()));
+        Assert.assertEquals("GET", span.tags().get(Tags.HTTP_METHOD.getKey()));
+        Assert.assertEquals("http://localhost:" + port + "/sync", span.tags().get(Tags.HTTP_URL.getKey()));
+        Assert.assertEquals(response.getStatusCode().value(), span.tags().get(Tags.HTTP_STATUS.getKey()));
+        Assert.assertEquals("java-spring-web-interceptor", span.tags().get(Tags.COMPONENT.getKey()));
+        Assert.assertEquals(0, span.logEntries().size());
+    }
+
+    @Test
+    public void testAsync() throws Exception {
+        {
```

Comment:
Actually I like this I can clearly see that nothing from inside a block is propagated to asserts.